crates/rpc-types-engine/src/payload.rs

```diff
@@ -0,0 +1,175 @@
+use alloy_primitives::{keccak256, Signature, B256};
+use alloy_rpc_types_engine::{
+    ExecutionPayload, ExecutionPayloadV1, ExecutionPayloadV2, ExecutionPayloadV3,
+};
+use derive_more::derive::{Display, From};
+use ssz::Decode;
+
+/// Optimism execution payload envelope in network format.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct OptimismNetworkExecutionPayloadEnvelope {
```

Comment:
This name is a little confusing, what is a more idiomatic name? 
essentially this is the CL network representation of unsafe blocks as they get gossiped by the sequencer.

Reply:
Tough. Maybe just `OptimismNetworkPayloadEnvelope`? No real need for the "execution" provision

Reply:
at some point we need to do a naming audit, because we currently use both `Optimism` and `Op` prefixes

Reply:
Noticed that as well... I should be able to take a pass at this early this week!

Reply:
https://github.com/alloy-rs/op-alloy/issues/150